pybamm/solvers/idaklu_solver.py

```diff
@@ -251,6 +275,49 @@ def resfn(t, y, inputs, ydot):
                 "mass_action", [v_casadi], [casadi.densify(mass_matrix @ v_casadi)]
             )
 
+            # if output_variables specified then convert functions to casadi
+            # expressions for evaluation within the idaklu solver
+            self.var_casadi_fcns = {}
+            self.var_casadi_idaklu_fcns = []
+            self.dvar_dy_fcns = []
+            self.dvar_dp_fcns = []
+            for key in self.output_variables:
+                # variable functions
+                if isinstance(
+                    model.variables_and_events[key], pybamm.ExplicitTimeIntegral
+                ):
+                    continue
```

Comment:
is ExplicitTimeIntegral not supported. Should we throw an error here?

Reply:
ExplicitTimeIntegral receives special treatment throughout the codebase. There is no action required here to convert this type of variable. This is confirmed by comparing the standard and new-solver outputs for these variables, which form part of the unit test cases in test_solvers/test_idaklu_solver.py#584-585.